File: query.cpp
#include <cstdio>
#include <string>
#include "HotelManager.h"

bool Query::quit() { return type == QUIT; }

Query Query::deserialize(char socketStream[]) {
	Query query;
	char userName[USERNAMESIZE];
	char bookID[BOOKIDSIZE];
	
	sprintf(socketStream, "%d%s%s%d%d%d%d%d%d%d",
		query.type, userName, bookID, query.hotelID,
		query.roomNum[0], query.roomNum[1], query.roomNum[2],
		query.checkin, query.checkout, query.price);
		
	query.userName = *(new std::string(userName));
	query.bookID = *(new std::string(bookID));
	
	return query;
}

File: Client/src/Query.java
public class Query implements java.io.Serializable, Cloneable
{
	private static final long serialVersionUID = 1L;
	
	public QueryType type;
	public String userName;
	public String bookID;
	public int hotelID;
	public int[] roomNum;
	public String checkin;
	public String checkout;
	/**
	 * 12/15
	 */
//	public String hotelAddress; 
	public int price;
	
	public Query() {
		roomNum = new int[3];
	}
	
	public Query clone() {
		try {
			return (Query)super.clone();
		} catch (CloneNotSupportedException e) {
			return null;
		}
	}
	
	public boolean quit() {
		return type == QueryType.QUIT;
	}
}
